### ISSUE TITLE
ui: Accurately capture statement performance for pipelined writes

### ISSUE THREAD

**Issue by piyush-singh** [cockroachdb/cockroach#36009]
When a transaction contains at least one write statement, the writes will now (as of 2.1) be pipelined: the db acknowledges a write almost as soon as the statement is issued (< 5 ms if gateway=leaseholder) while the consensus replication is carried out behind the scenes in the transaction. Any waiting on consensus happens on commit rather than synchronously when the statement is issued.

In the statements page, an unsuspecting user might see the statement performance and think their performance is incredible, when in actuality they are not seeing the larger picture of their complete transaction performance.

We should capture the wait that happens in the commit phase of the transaction (perhaps per transaction fingerprint) and report that back to the user so that they will have a more accurate picture of true performance.

cc @awoods187

**Comment by nvanbenschoten** [comment]
I think we should reframe this a little bit, as doing so might help us come to a better solution. Post-transaction pipelining, replication latency is no longer a property of statements. Instead, it is a property of transactions. So I think the way to look at this is that each statement takes as long as it takes and then the statement's transaction adds its own overhead. Implicit transactions are a single statement within a transaction and they still fit in to this model.

With that in mind, a solution like trying to associate replication latency with a specific statement seems backwards. But that's not what you're suggesting, so I think we're on the same page :)

**Comment by awoods187** [comment]
Can you post a screen shot of what you are seeing and where it's confusing?

**Comment by jseldess** [comment]
I almost released a demo video of geo-partitioning with exactly this misunderstanding of the statements page. See 8:15: https://drive.google.com/file/d/1kt5gZXVefnSBLxTGW7yFr9VHcYYATuJQ/view?usp=sharing

**Comment by nvanbenschoten** [comment]
We discussed this issue in the severe issues group yesterday. A short term fix that would alleviate this issue would be to surface whether statements are part of implicit transactions or not. These statements should not be grouped together. This could be as simple as an `Implicit Transaction` column on the statements page and then a new row in the same section as "Used cost-based optimizer?" on the statement detail page.

Does that sound workable for 19.1 @piyush-singh?

**Comment by piyush-singh** [comment]
cc @celiala let's discuss if this is manageable

**Comment by piyush-singh** [comment]
Jordan is going to help us by setting up the backend/crdb_internal table appropriately. Issue: #36286

**Comment by celiala** [comment]
Once #36341 lands, work on the CRDB side to surface 'Implicit Transaction' flag on Statements Page will be complete (should merge shortly).

However, we probably want to update the reg cluster to separate out implicit & explicit transactions. Opening separate issue on registration repo [#212](https://github.com/cockroachlabs/registration/issues/212) to track this.
